examples/curl-report.rs

```diff
@@ -0,0 +1,103 @@
+use threescalers::api_call::*;
+use threescalers::application::*;
+use threescalers::credentials::*;
+use threescalers::extensions::Extensions;
+use threescalers::http::request::FromRequest;
+use threescalers::http::Request;
+use threescalers::service::*;
+use threescalers::transaction::Transaction;
+use threescalers::usage::Usage;
+
+use threescalers::http::request::curl::CurlEasyClient;
+
+use curl::easy::{Easy, Transfer};
+
+fn main() -> Result<(), threescalers::errors::Error> {
+    let creds = Credentials::ServiceToken(ServiceToken::from("12[3]token"));
+    let svc = Service::new("svc123", creds);
+    let uks = [
+        "userkey_1",
+        "userkey_2",
+        "userkey_3",
+        "userkey 4",
+        "userkey 5",
+    ];
+    let apps = uks
+        .into_iter()
+        .map(|uk| Application::from(UserKey::from(*uk)))
+        .collect::<Vec<_>>();
+
+    println!("Apps: {:#?}", apps);
+
+    let usages = [
+        ("metric11", 11),
+        ("metric12", 12),
+        ("metric21", 21),
+        ("metric22", 22),
+        ("metric31", 31),
+        ("metric32", 32),
+        ("metric41", 41),
+        ("metric42", 42),
+        ("metric51", 51),
+        ("metric52", 52),
+    ]
+    .chunks(2)
+    .map(|metrics_and_values| Usage::new(metrics_and_values))
+    .collect::<Vec<_>>();
+
+    println!("Usages: {:#?}", usages);
+
+    let ts = Default::default();
+
+    let txns = apps
+        .iter()
+        .zip(&usages)
+        .map(|(a, u)| Transaction::new(a, None, Some(u), Some(&ts)))
+        .collect::<Vec<_>>();
+
+    let mut extensions = Extensions::new();
+    extensions.insert("no_body", "1");
+    extensions.insert("testing[=]", "0[=:=]0");
+    let mut apicall = ApiCall::builder(&svc);
+    let apicall = apicall
+        .transactions(&txns)
+        .extensions(&extensions)
+        .kind(Kind::Report)
+        .build()?;
+    let request = Request::from(&apicall);
+
+    println!("apicall: {:#?}", apicall);
+    println!("request: {:#?}", request);
+
+    let _ = run_request(request);
+
+    Ok(())
+}
+
+fn run_request(request: Request) -> Result<(), curl::Error> {
+    let mut client = Easy::new();
+    let _ = client.verbose(true).unwrap();
+    let curlclient =
+        CurlEasyClient::from_request(request, (&mut client, "https://echo-api.3scale.net"));
```

Comment:
This example is very similar to the one that uses reqwest. It made me think that it might be helpful to define a `Client` trait with `from_request`, `exec_request` and `show_response`. Something to think about if we need to add more examples like this one.

Reply:
for the examples, yes, and it might be useful to have some sort of fixtures for adding all the data about transactions, apps, service credentials, etc.

Reply:
:+1: